src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationReaderILGen.cs

```diff
@@ -3046,34 +3046,78 @@ private void WriteElement(string source, string arrayName, string choiceSource,
                 {
                 }
 
-                WriteSourceBegin(source);
-                if (element.Mapping.TypeDesc == QnameTypeDesc)
+                if (element.Mapping.TypeDesc.Type == typeof(TimeSpan))
                 {
-                    MethodInfo XmlSerializationReader_ReadElementQualifiedName = typeof(XmlSerializationReader).GetMethod(
-                           "ReadElementQualifiedName",
-                           CodeGenerator.InstanceBindingFlags,
-                           Array.Empty<Type>()
-                           );
+                    MethodInfo XmlSerializationReader_get_Reader = typeof(XmlSerializationReader).GetMethod(
+                       "get_Reader",
+                       CodeGenerator.InstanceBindingFlags,
+                       Array.Empty<Type>()
+                       );
+                    MethodInfo XmlReader_get_IsEmptyElement = typeof(XmlReader).GetMethod(
+                        "get_IsEmptyElement",
+                        CodeGenerator.InstanceBindingFlags,
+                        Array.Empty<Type>()
+                        );
                     ilg.Ldarg(0);
-                    ilg.Call(XmlSerializationReader_ReadElementQualifiedName);
+                    ilg.Call(XmlSerializationReader_get_Reader);
+                    ilg.Call(XmlReader_get_IsEmptyElement);
+                    ilg.If();
+                    WriteSourceBegin(source);
+                    MethodInfo XmlReader_Skip = typeof(XmlReader).GetMethod(
+                        "Skip",
+                        CodeGenerator.InstanceBindingFlags,
+                        Array.Empty<Type>()
+                        );
+                    ilg.Ldarg(0);
+                    ilg.Call(XmlSerializationReader_get_Reader);
+                    ilg.Call(XmlReader_Skip);
+                    ConstructorInfo TimeSpan_ctor = typeof(TimeSpan).GetConstructor(
```

Comment:
The IL you're generating is equivalent to `value = new TimeSpan(0);` which actually does more work than `value = default(TimeSpan)`. Using default on a struct allocates space on the stack and ensures the bytes needed are zeroed out. This happens anyway when you use `new` but it also makes a method call, which in this case also assigns the passed in 0 value to the memory allocated for the TimeSpan. I'm curious why you chose this way to implement the functionality instead of the equivalent to using `default`.

Reply:
@mconnew These are copied here from full framework.